### PR TITLE
Migrate the public endpoint Delete DAG to FastAPI

### LINKED ISSUE #42650: AIP-84 Migrate the public endpoint Delete DAG to FastAPI
### Description

The Delete DAG public endpoint is at `api_connexion/endpoints/dag_endpoint.py` under `delete_dag`. We need to migrate it to the `api_fastapi/views/public/dags.py` under a `delete_dag` or similar.

Features and functionality of the endpoint will remain unchanged.

### Use case/motivation

AIP-84 cWiki Link: https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-84+UI+REST+API

### Related issues

Related to: https://github.com/apache/airflow/issues/42370

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by omkar-foss**
closes: #42650
related: #42370

This migrates the Delete DAG API from `api_connexion` to `api_fastapi`, with a supporting test.

### CODE REVIEW

**Comment by omkar-foss** [comment]
Merge conflict resolved, PR is ready for review.

**Comment by omkar-foss** [comment]
Tests failing after adding migration marker for legacy APIs. Could someone please add the `legacy api` label to this PR? Thanks.

**Comment by pierrejeambrun** [comment]
Label added :)

**Comment by omkar-foss** [comment]
The mypy tests for `mark_fastapi_migration_done` had been failing, so have updated it's signature to be more tolerant [here](https://github.com/apache/airflow/pull/42914/files#diff-0540fd6b9fb773735b6a583f8637809d4b53253c6415863c20634ab80e2f1a3bR31). Tests passing now.

File: airflow/utils/api_migration.py [hunk 1]
```diff
@@ -25,13 +25,10 @@
...
 from typing import Callable, TypeVar
 
-from airflow.typing_compat import ParamSpec
-
-PS = ParamSpec("PS")
 RT = TypeVar("RT")
 
 
-def mark_fastapi_migration_done(function: Callable[PS, RT]) -> Callable[PS, RT]:
+def mark_fastapi_migration_done(function: Callable[..., RT]) -> Callable[..., RT]:
```
Comment (pierrejeambrun) [inline airflow/utils/api_migration.py:34]: Do you mind elaborating why this is needed ?

**Review by pierrejeambrun — DISMISSED** [review]

File: airflow/utils/api_migration.py (hunk 1 above)
Comment (omkar-foss) [reply airflow/utils/api_migration.py:34]: Yes sure, have added `...` in place or `PS` to accept any param, as the mypy tests had been failing after `mark_fastapi_migration_done` was used [here](https://github.com/apache/airflow/pull/42914/files#diff-bf872a9f63486e4349959ddd97e607b9779221bd15a049fd1b94e9276d2d0bf5R219), with the error as given below.

If there's a better way to fix this, please let me know, will update PR accordingly. Thanks!

```bash
airflow/api_connexion/endpoints/dag_endpoint.py:217: error: Argument 1 to
"mark_fastapi_migration_done" has incompatible type
"Union[Callable[[str, Any], Union[Response, tuple[object, int], Mapping[str, Any]]], Callable[..., Any]]";
expected "Callable[[VarArg(str), KwArg(Any)], Any]"  [arg-type]
    @mark_fastapi_migration_done
     ^
Found 1 error in 1 file (checked 1046 source files)
Error 1 returned
```

Comment (pierrejeambrun) [reply airflow/utils/api_migration.py:34]: Got it thanks.

**Review by pierrejeambrun — APPROVED** [review]
Thanks

**Pull request merged**